ccl/partitionccl: TestGenerateSubzoneSpans failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/91b11dae416f3d9b55fadd2a4b096e94d874176c

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestGenerateSubzoneSpans PKG=github.com/cockroachdb/cockroach/pkg/ccl/partitionccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1042784&tab=buildLog

```

--- FAIL: TestGenerateSubzoneSpans (0.00s)
panic: value type unknown cannot be used for table columns [recovered]
	panic: value type unknown cannot be used for table columns [recovered]
	panic: value type unknown cannot be used for table columns

goroutine 3157 [running]:
testing.tRunner.func1(0xc4218fa870)
	/usr/local/go/src/testing/testing.go:742 +0x29d
panic(0x2af1b60, 0xc4212e8780)
	/usr/local/go/src/runtime/panic.go:502 +0x229
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:92 +0x432
panic(0x2af1b60, 0xc4212e8780)
	/usr/local/go/src/runtime/panic.go:502 +0x229
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FmtCtx).FormatNode(0xc4218a42d0, 0x3194180, 0xc4211e5650)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:295 +0x4ac
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.AsStringWithFlags(0x3194180, 0xc4211e5650, 0x12000, 0x31fa300, 0x3194180)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:305 +0x54
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.Serialize(0x3194180, 0xc4211e5650, 0xc4211e5650, 0x3194180)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:323 +0x3e
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.allPartitioningTests(0xc4218e7da0, 0xea35a10d81e2ea4f, 0xc420d3ad60, 0xc4209bcf98)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/partition_test.go:871 +0x4b34
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.TestGenerateSubzoneSpans(0xc4218fa870)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/zone_test.go:288 +0x77
testing.tRunner(0xc4218fa870, 0x2d74d20)
	/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x2e0

goroutine 1 [chan receive]:
testing.(*T).Run(0xc4218fa870, 0x2c58771, 0x18, 0x2d74d20, 0x976d01)
	/usr/local/go/src/testing/testing.go:825 +0x301
testing.runTests.func1(0xc420189590)
	/usr/local/go/src/testing/testing.go:1063 +0x64
testing.tRunner(0xc420189590, 0xc4207c3dd8)
	/usr/local/go/src/testing/testing.go:777 +0xd0
testing.runTests(0xc4202fa3c0, 0x4424dc0, 0x8, 0x8, 0xbefa4d46b90a63cb)
	/usr/local/go/src/testing/testing.go:1061 +0x2c4
testing.(*M).Run(0xc4204cc280, 0x0)
	/usr/local/go/src/testing/testing.go:978 +0x171
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.TestMain(0xc4204cc280)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/main_test.go:31 +0xda
main.main()
	_testmain.go:54 +0x151

goroutine 6 [syscall]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:139 +0xa6
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 7 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1203 +0xf1
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:619 +0x126

goroutine 8 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:626 +0xab
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:620 +0x13e

goroutine 51 [select, locked to thread]:
runtime.gopark(0x2d79ef8, 0x0, 0x2c331ad, 0x6, 0x18, 0x1)
	/usr/local/go/src/runtime/proc.go:291 +0x11a
runtime.selectgo(0xc420499f50, 0xc4204b00c0)
	/usr/local/go/src/runtime/select.go:392 +0xe50
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal_unix.go:549 +0x1f4
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2361 +0x1

goroutine 67 [select]:
github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.(*worker).start(0xc420370540)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view/worker.go:144 +0x11d
created by github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.init.0
```

SHA: https://github.com/cockroachdb/cockroach/commits/81ed730f073675fe5d634274658e1ff5e291157d

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestGenerateSubzoneSpans PKG=github.com/cockroachdb/cockroach/pkg/ccl/partitionccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1046687&tab=buildLog

```

--- FAIL: TestGenerateSubzoneSpans (0.00s)
panic: value type unknown cannot be used for table columns [recovered]
	panic: value type unknown cannot be used for table columns [recovered]
	panic: value type unknown cannot be used for table columns

goroutine 3321 [running]:
testing.tRunner.func1(0xc42017d770)
	/usr/local/go/src/testing/testing.go:742 +0x567
panic(0x3996500, 0xc4208be300)
	/usr/local/go/src/runtime/panic.go:502 +0x24a
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:91 +0x526
panic(0x3996500, 0xc4208be300)
	/usr/local/go/src/runtime/panic.go:502 +0x24a
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FmtCtx).FormatNode(0xc420bd8240, 0x435d600, 0xc420f1ca50)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:295 +0x660
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.AsStringWithFlags(0x435d600, 0xc420f1ca50, 0x12000, 0x3787aa0, 0x3a3ef00)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:305 +0x66
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.Serialize(0x435d600, 0xc420f1ca50, 0xc420f1ca50, 0x435d600)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:323 +0x4c
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.allPartitioningTests(0xc420f1c090, 0xb5205565e2a303a4, 0xc420308420, 0xc42028cf68)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/partition_test.go:870 +0x6661
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.TestGenerateSubzoneSpans(0xc42017d770)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/zone_test.go:288 +0x88
testing.tRunner(0xc42017d770, 0x3c1a668)
	/usr/local/go/src/testing/testing.go:777 +0x16e
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x565

goroutine 1 [chan receive]:
testing.(*T).Run(0xc42017d590, 0x3afe010, 0x18, 0x3c1a668, 0xc4207f3c01)
	/usr/local/go/src/testing/testing.go:825 +0x597
testing.runTests.func1(0xc42017d590)
	/usr/local/go/src/testing/testing.go:1063 +0xa5
testing.tRunner(0xc42017d590, 0xc4207f3d68)
	/usr/local/go/src/testing/testing.go:777 +0x16e
testing.runTests(0xc4202de320, 0x5789ba0, 0x8, 0x8, 0xbefaf7161683a71e)
	/usr/local/go/src/testing/testing.go:1061 +0x4e2
testing.(*M).Run(0xc420478080, 0x0)
	/usr/local/go/src/testing/testing.go:978 +0x2ce
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.TestMain(0xc420478080)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/main_test.go:31 +0x180
main.main()
	_testmain.go:54 +0x22b

goroutine 20 [syscall]:
os/signal.signal_recv(0x9a8b51)
	/usr/local/go/src/runtime/sigqueue.go:139 +0xa6
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x30
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x4f

goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1204 +0x132
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:620 +0x1d4

goroutine 6 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:627 +0xf2
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:621 +0x1ec

goroutine 38 [select, locked to thread]:
runtime.gopark(0x3c1fa60, 0x0, 0x3ad86ca, 0x6, 0x18, 0x1)
	/usr/local/go/src/runtime/proc.go:291 +0xf9
runtime.selectgo(0xc420293f50, 0xc4200c82a0)
	/usr/local/go/src/runtime/select.go:392 +0x11d4
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal_unix.go:549 +0x19f
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2361 +0x1

goroutine 9 [select]:
github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.(*worker).start(0xc4203c8880)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view/worker.go:144 +0x18f
created by github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.init.0
```

SHA: https://github.com/cockroachdb/cockroach/commits/1cd7dcfe1bbf487a714c7f00a5dd709360dbe9e4

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestGenerateSubzoneSpans PKG=github.com/cockroachdb/cockroach/pkg/ccl/partitionccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1049186&tab=buildLog

```

--- FAIL: TestGenerateSubzoneSpans (0.00s)
panic: value type unknown cannot be used for table columns [recovered]
	panic: value type unknown cannot be used for table columns [recovered]
	panic: value type unknown cannot be used for table columns

goroutine 3224 [running]:
testing.tRunner.func1(0xc4216460f0)
	/usr/local/go/src/testing/testing.go:742 +0x29d
panic(0x2b78140, 0xc420e283c0)
	/usr/local/go/src/runtime/panic.go:502 +0x229
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:91 +0x432
panic(0x2b78140, 0xc420e283c0)
	/usr/local/go/src/runtime/panic.go:502 +0x229
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FmtCtx).FormatNode(0xc42136e120, 0x32172a0, 0xc4216b8c30)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:295 +0x4ac
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.AsStringWithFlags(0x32172a0, 0xc4216b8c30, 0x12000, 0x327d400, 0x32172a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:305 +0x54
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.Serialize(0x32172a0, 0xc4216b8c30, 0xc4216b8c30, 0x32172a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:323 +0x3e
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.allPartitioningTests(0xc4216b81e0, 0xb2f1a5c2eaad8ba1, 0xc4203b4540, 0xc420cd7f98)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/partition_test.go:870 +0x4b34
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.TestGenerateSubzoneSpans(0xc4216460f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/zone_test.go:288 +0x77
testing.tRunner(0xc4216460f0, 0x2dfa680)
	/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x2e0

goroutine 1 [chan receive]:
testing.(*T).Run(0xc4216460f0, 0x2cde526, 0x18, 0x2dfa680, 0x9c6201)
	/usr/local/go/src/testing/testing.go:825 +0x301
testing.runTests.func1(0xc4201594a0)
	/usr/local/go/src/testing/testing.go:1063 +0x64
testing.tRunner(0xc4201594a0, 0xc4207d1dd8)
	/usr/local/go/src/testing/testing.go:777 +0xd0
testing.runTests(0xc4202cf720, 0x44c3c80, 0x8, 0x8, 0xbefbf315db56ef5d)
	/usr/local/go/src/testing/testing.go:1061 +0x2c4
testing.(*M).Run(0xc42017ec00, 0x0)
	/usr/local/go/src/testing/testing.go:978 +0x171
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.TestMain(0xc42017ec00)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/main_test.go:31 +0xda
main.main()
	_testmain.go:54 +0x151

goroutine 6 [syscall]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:139 +0xa6
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 7 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1204 +0xf1
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:620 +0x126

goroutine 8 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:627 +0xab
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:621 +0x13e

goroutine 36 [select, locked to thread]:
runtime.gopark(0x2dff930, 0x0, 0x2cb8e7c, 0x6, 0x18, 0x1)
	/usr/local/go/src/runtime/proc.go:291 +0x11a
runtime.selectgo(0xc420481750, 0xc4204a6060)
	/usr/local/go/src/runtime/select.go:392 +0xe50
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal_unix.go:549 +0x1f4
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2361 +0x1

goroutine 23 [select]:
github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.(*worker).start(0xc420464700)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view/worker.go:144 +0x11d
created by github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.init.0
```

SHA: https://github.com/cockroachdb/cockroach/commits/859214b81838a4ba33048b81497442ce5774baa7

Parameters:
```
TAGS=
GOFLAGS=
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestGenerateSubzoneSpans PKG=github.com/cockroachdb/cockroach/pkg/ccl/partitionccl TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1053322&tab=buildLog

```

--- FAIL: TestGenerateSubzoneSpans (0.00s)
panic: value type unknown cannot be used for table columns [recovered]
	panic: value type unknown cannot be used for table columns [recovered]
	panic: value type unknown cannot be used for table columns

goroutine 3025 [running]:
testing.tRunner.func1(0xc4215380f0)
	/usr/local/go/src/testing/testing.go:742 +0x29d
panic(0x2bd78e0, 0xc4216f6a20)
	/usr/local/go/src/runtime/panic.go:502 +0x229
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:91 +0x432
panic(0x2bd78e0, 0xc4216f6a20)
	/usr/local/go/src/runtime/panic.go:502 +0x229
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.(*FmtCtx).FormatNode(0xc420f66120, 0x3279180, 0xc421404db0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:295 +0x4ac
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.AsStringWithFlags(0x3279180, 0xc421404db0, 0x12000, 0x32e0f00, 0x3279180)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:305 +0x54
github.com/cockroachdb/cockroach/pkg/sql/sem/tree.Serialize(0x3279180, 0xc421404db0, 0xc421404db0, 0x3279180)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/sem/tree/format.go:323 +0x3e
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.allPartitioningTests(0xc4214045d0, 0xe86cd51cfc2c1863, 0xc4217ce500, 0xc420df2f98)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/partition_test.go:870 +0x4b34
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.TestGenerateSubzoneSpans(0xc4215380f0)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/zone_test.go:288 +0x77
testing.tRunner(0xc4215380f0, 0x2e58368)
	/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:824 +0x2e0

goroutine 1 [chan receive]:
testing.(*T).Run(0xc4215380f0, 0x2d41697, 0x18, 0x2e58368, 0x9c6201)
	/usr/local/go/src/testing/testing.go:825 +0x301
testing.runTests.func1(0xc420824000)
	/usr/local/go/src/testing/testing.go:1063 +0x64
testing.tRunner(0xc420824000, 0xc4205f5dd8)
	/usr/local/go/src/testing/testing.go:777 +0xd0
testing.runTests(0xc420428d60, 0x4560d00, 0x8, 0x8, 0xbefc9b8da027ee4c)
	/usr/local/go/src/testing/testing.go:1061 +0x2c4
testing.(*M).Run(0xc420546200, 0x0)
	/usr/local/go/src/testing/testing.go:978 +0x171
github.com/cockroachdb/cockroach/pkg/ccl/partitionccl.TestMain(0xc420546200)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/partitionccl/main_test.go:31 +0xda
main.main()
	_testmain.go:54 +0x151

goroutine 6 [syscall]:
os/signal.signal_recv(0x0)
	/usr/local/go/src/runtime/sigqueue.go:139 +0xa6
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 7 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.flushDaemon()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1204 +0xf1
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:620 +0x126

goroutine 8 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.signalFlusher()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:627 +0xab
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:621 +0x13e

goroutine 34 [select, locked to thread]:
runtime.gopark(0x2e5d660, 0x0, 0x2d1be1c, 0x6, 0x18, 0x1)
	/usr/local/go/src/runtime/proc.go:291 +0x11a
runtime.selectgo(0xc420468750, 0xc420462060)
	/usr/local/go/src/runtime/select.go:392 +0xe50
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal_unix.go:549 +0x1f4
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2361 +0x1

goroutine 67 [select]:
github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.(*worker).start(0xc4207f8240)
	/go/src/github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view/worker.go:144 +0x11d
created by github.com/cockroachdb/cockroach/vendor/go.opencensus.io/stats/view.init.0
```

--- a/pkg/sql/sem/tree/format.go
+++ b/pkg/sql/sem/tree/format.go
@@ -292,7 +292,7 @@ func (ctx *FmtCtx) FormatNode(n NodeFormatter) {
 			ctx.WriteString(":::")
 			colType, err := coltypes.DatumTypeToColumnType(typ)
 			if err != nil {
-				panic(err)
+				panic(fmt.Sprintf("invalid datatype %v", typ))
 			}
 			colType.Format(ctx.Buffer, f.EncodeFlags())
 		}

--- a/pkg/sql/sqlbase/testutils.go
+++ b/pkg/sql/sqlbase/testutils.go
@@ -196,17 +196,7 @@ func RandDatumWithNullChance(rng *rand.Rand, typ ColumnType, nullChance int) tre
 		return tree.DNull
 	case ColumnType_ARRAY:
 		if typ.ArrayContents == nil {
-			var contentsTyp ColumnType
-		LOOP:
-			for {
-				contentsTyp = RandColumnType(rng)
-				switch contentsTyp.SemanticType {
-				// Can't have an array of an array.
-				case ColumnType_ARRAY, ColumnType_JSONB:
-				default:
-					break LOOP
-				}
-			}
+			var contentsTyp = RandArrayContentsColumnType(rng)
 			typ.ArrayContents = &contentsTyp.SemanticType
 			typ.Locale = contentsTyp.Locale
 		}
@@ -260,7 +250,17 @@ func RandCollationLocale(rng *rand.Rand) *string {
 
 // RandColumnType returns a random ColumnType value.
 func RandColumnType(rng *rand.Rand) ColumnType {
-	typ := ColumnType{SemanticType: columnSemanticTypes[rng.Intn(len(columnSemanticTypes))]}
+	return randColumnType(rng, columnSemanticTypes)
+}
+
+// RandArrayContentsColumnType returns a random ColumnType that's guaranteed
+// to be valid to use as the contents of an array.
+func RandArrayContentsColumnType(rng *rand.Rand) ColumnType {
+	return randColumnType(rng, arrayElemSemanticTypes)
+}
+
+func randColumnType(rng *rand.Rand, types []ColumnType_SemanticType) ColumnType {
+	typ := ColumnType{SemanticType: types[rng.Intn(len(types))]}
 	if typ.SemanticType == ColumnType_BIT {
 		typ.Width = int32(rng.Intn(50))
 	}
@@ -268,12 +268,12 @@ func RandColumnType(rng *rand.Rand) ColumnType {
 		typ.Locale = RandCollationLocale(rng)
 	}
 	if typ.SemanticType == ColumnType_ARRAY {
-		typ.ArrayContents = &arrayElemSemanticTypes[rng.Intn(len(arrayElemSemanticTypes))]
-		if *typ.ArrayContents == ColumnType_COLLATEDSTRING {
+		inner := RandArrayContentsColumnType(rng)
+		if inner.SemanticType == ColumnType_COLLATEDSTRING {
 			// TODO(justin): change this when collated arrays are supported.
-			s := ColumnType_STRING
-			typ.ArrayContents = &s
+			inner.SemanticType = ColumnType_STRING
 		}
+		typ.ArrayContents = &inner.SemanticType
 	}
 	if typ.SemanticType == ColumnType_TUPLE {
 		// Generate tuples between 0 and 4 datums in length

